tesseract/tesseract_collision/include/tesseract_collision/core/common.h

```diff
@@ -119,7 +119,8 @@ inline ContactResult* processResult(ContactTestData& cdata,
   if (cdata.req.is_valid && !cdata.req.is_valid(contact))
     return nullptr;
 
-  if (!(contact.distance < cdata.collision_margin_data.getPairCollisionMarginData(key.first, key.second)))
+  if ((cdata.req.calculate_distance || cdata.req.calculate_penetration) &&
```

Comment:
Only look up pair distance if distance or penetration data was requested.

Reply:
This looks fine. I think what I was thinking before is that we need to add this before these lines (at 121) to keep from doing the lookup when the type is FIRST and we already found one.

```
  if (found && cdata.req.type == ContactTestType::FIRST)
    return nullptr;
```

Reply:
It should not enter this function a second time if the test type is `ContactTestType::FIRST`. It sets the `done` flag which prevents any further contact checks. I have an assert below which checks to make sure this does not happen `assert(cdata.req.type != ContactTestType::FIRST);`.

Reply:
This is fine then.